### PR TITLE
Feat/marketplace logic

### PR DESCRIPTION
**Pull request by suwandre**
@shonsirsha Here are some important updates regarding the changes I want to implement. If everything is okay from your side, let's merge it.

**NOTE**: ALL of the below mentioned changes do **NOT** have security middleware implemented (we can implement this later on, just FYI).

1. `getItemsOnSale` and `getItemOnSale` is created on `marketplaceLogic.js`. I'm still unsure how slow or fast this will be, so we do need testing. But the implementation is completed :)

2. `deleteItemOnSale` is also implemented. Not sure if you also prefer to have `cancelItemOnSale` which will then call `deleteItemOnSale`, but this can be talked upon.

3. `ItemsOnSale` class on Moralis now has `duration` which is calculated by `endTime` (given by the user from the FE) minus the current time.

4. **NOT RELATED TO MARKETPLACE** `getGenesisNBMon` has now been updated to take data from Moralis. I commented the blockchain part just in case we would still need it (either for benchmark or feasibility purposes). Minting and hatching now includes adding/updating into the `Genesis_NBMons` class which I created on Moralis to store, which the `getGenesisNBMon` endpoint obtains its data from.

### CODE REVIEW

File: routes/marketplace.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        endingPrice, 
+        minimumReserveBid, 
+        duration
+        )
+        .catch((err) => res.json(err.message));
+    
+    res.json(addItem);
+});
+
+router.get("getItems", async (_, res) => {
```
Comment (shonsirsha) [inline routes/marketplace.js:41]: Missing `/` infront of the route name

File: routes/marketplace.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    let items = await getItemsOnSale()
+        .catch((err) => res.json(err.message));
+
+    res.json(items);
+});
+
+router.get("getItem", async (req, res) => {
```
Comment (shonsirsha) [inline routes/marketplace.js:48]: Missing / infront of the route name

File: routes/marketplace.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    const { tokenId } = req.body;
+    let item = await getItemOnSale(tokenId).catch((err) => res.json(err.message));
+
+    res.json(item);
+});
+
+router.post("generateTxSalt", async (_, res) => {
```
Comment (shonsirsha) [inline routes/marketplace.js:55]: Missing / infront of the route name

File: routes/marketplace.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    let salt = generateTxSalt();
+
+    res.json(salt);
+});
+
+router.post("deleteItem", async (req, res) => {
```
Comment (shonsirsha) [inline routes/marketplace.js:61]: Missing / infront of the route name

File: routes/genesisNBMonHatching.js [hunk 5]
```diff
@@ -55,4 +56,13 @@ router.post("/uploadHatchedMetadata", async (req, res) => {
 	res.json(upload);
 });
 
+router.post("/updateHatchedNBMon", async (req, res) => {
+	const {id, stringMetadata, numericMetadata, boolMetadata} = req.body;
```
Comment (shonsirsha) [inline routes/genesisNBMonHatching.js:60]: Imo it's better to have the FE to only send `id` and not the metadata. So the way this endpoint would work is to then it should look up that `id` from blockchain and fetch all the needed metadata from there.

To get nbmon data from blockchain I believe you still have the code commented out in this PR - so just uncomment that, extract it into its own function, and call it here. :)

File: logic/genesisNBMonLogic.js [hunk 6]
```diff
@@ -13,6 +13,7 @@ const {
 	getAttackEffectiveness,
 	getDefenseEffectiveness,
 } = require("../logic/typeEffectivenessLogic");
+const { query } = require("express");
```
Comment (shonsirsha) [inline logic/genesisNBMonLogic.js:16]: I believe this isn't used. You probably imported it by accident - so you could remove it

**Review by shonsirsha — CHANGES_REQUESTED** [review]
Please address the changes before merging :) Also please pull from dev before you start making changes.

**Comment by suwandre** [comment]
Everything else has been updated apart from the /updateHatchedNBMon change. Once we update the RPC URL to Matic, I'll keep this in mind and change it asap. Thanks!

**Pull request merged**